src/circuits/safe_accumulator.rs

```diff
@@ -0,0 +1,115 @@
+use super::super::chips::safe_accumulator::{SafeAccumulatorConfig, SafeACcumulatorChip};
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*};
+
+#[derive(Default)]
+struct SafeAccumulatorCircuit {
+    pub values: Vec<Value<Fp>>,
+    pub accumulated_value: [Value<Fp>; 4],
```

Comment:
It doesn't seem intuitive to me that the `accumulated_value` array has to be passed to instantiate the circuit. An idea could be to refactor the  assignment functions of the chip as follows:

- `assign_first_row` only assign 0 to the various accumulator columns and returns `AssignedCells`
- `assign` takes as input the value to be added to the accumulator `Value<F>` and the `AssignedCells` from the previous round, and returns `AssignedCells`.

Reply:
I understand that it

> * `assign_first_row` only assign 0 to the various accumulator columns and returns `AssignedCells`

I mention above https://github.com/summa-dev/halo2-experiments/pull/5#discussion_r1182477175. I want to show none zero-initialized accumulator. how about a way of initializing the accumulator with instance value instead?

> * `assign` takes as input the value to be added to the accumulator `Value<F>` and the `AssignedCells` from the previous round, and returns `AssignedCells`.

Yes, I refactored at [feat: check public value at last accumulate cells](https://github.com/summa-dev/halo2-experiments/pull/5/commits/ee1b551835460cdb0e90d1d340f9fd0a337279d4)